File: Coding Challenge 10/morse_code.rb
#example input: morse_code("cat in hat")
MORSE_CODE = {
  "a" => ".-",
  "b" => "-...",
  "c" => "-.-.",
  "d" => "-..",
  "e" => ".",
  "f" => "..-.",
  "g" => "--.",
  "h" => "....",
  "i" => "..",
  "j" => ".---",
  "k" => "-.-",
  "l" => ".-..",
  "m" => "--",
  "n" => "-.",
  "o" => "---",
  "p" => ".--.",
  "q" => "--.-",
  "r" => ".-.",
  "s" => "...",
  "t" => "-",
  "u" => "..-",
  "v" => "...-",
  "w" => ".--",
  "x" => "-..-",
  "y" => "-.--",
  "z" => "--.."
}

def morse_code(str)
  words = str.split(" ")
  coded_words = words.map { |word| morse_code_word(word) }
  coded_words.join("  ")
end

def morse_code_word(word)
  letters = word.split("")
  codes = letters.map { |l| MORSE_CODE[l] }
  codes.join(" ")
end

File: Coding Challenge 10/morse_code.m
function result=morse_code(str)
    str=strread(str,'%s','delimiter',' ');
    result=[];
    for i=1:length(str)
        result=[result '  ' morse_code_word(str{i})];
    end
    
end


function result=morse_code_word(word)
    letters='abcdefghijklmnopqrstuvwxyz';
    code={'.-' '-...' '-.-.' '-..' '.' '..-.' '--.' '....' '..' '.---' '-.-.' '.-..' '--' '-.' '---' '.--.' '--.-' '.-.' '...' '-' '..-' '...-' '.--' '-..-' '-.--' '--..'};
    result=[];
    for i=1:length(word)
        I=find(letters==word(i));
        result=[result ' ' code{I}];
    end
end